numpy/core/shape_base.py

```diff
@@ -21,6 +21,13 @@ def _atleast_1d_dispatcher(*arys):
     return arys
 
 
+def _atleast_nd_result(arys, ndim, pos, unpack=True):
+    lst = [atleast_nd(a, ndim, pos) for a in arys]
+    if unpack and len(lst) == 1:
+        return lst[0]
+    return lst
```

Comment:
I'd be tempted to tweak this slightly, to
```suggestion
def _unpack(args):
    if len(args) == 1:
        return args[0]
    return args
```

and then below implement `atleast_1d` as `return _unpack([atleast_nd(a, 1, 0) for a in args])`, etc

Reply:
I specifically didn't want to write the loop over and over, so will make a hybrid with your idea, because it's definitely neater than having to parse and extra `unpack` parameter.